psi-ms.obo

```diff
@@ -21985,6 +21985,53 @@ is_a: MS:1003201 ! library provenance attribute
 xref: value-type:xsd:string "The allowed value-type for this CV term."
 relationship: has_value_type xsd\:string ! The allowed value-type for this CV term
 
+[Term]
+id: MS:1003207
+name: library creation software
+def: "Library creation software" [PSI:MS]
+is_a: MS:1000531 ! software
+is_a: MS:1003171 ! spectral library attribute
+relationship: has_value_type xsd\:string ! The allowed value-type for this CV term
```

Comment:
Are we not encouraging software vendors to add their software as CV terms and using those whenever possible instead of free text?

Reply:
I had that question too, but the entry from the CV terms to create table had included this value type expectation.

I think registration is preferable to free text for consumers of CV term containing documents, but for new tool writers it is a chore.

Reply:
Yes it's a chore but it's also the status quo. It's going to make me twitch if some of the PSI formats only support free text and others don't. :stuck_out_tongue:

Reply:
I think I remember the issue now. With mzSpecLib, the design allows you to have one CV param whose value is another CV param. This is because mzSpecLib is structured almost entirely by CV params instead of a formal schema breaking down the data model.

In XML, we'd write:
```xml
<LibraryCreationSoftware version="1.2.3" id="creation_software1">
    <cvParam accession="MS:1001477"  name="SpectraST" />
</LibraryCreationSoftware>
``` 
In mzSpecLib, that could be constructed like this:
```ps1
MS:1003207|library creation software=SpectraST
```
or like this (redundantly?):
```ps1
MS:1003207|library creation software=MS:1001477|SpectraST
```
Here though the value is a term, not a string, and we haven't formally stipulated that situation yet.

However, if we weren't insisting on the key naming the scope and just hope we can tell by context as we can in the narrowed context in an XML file, we could just write:
```ps1
MS:1001477|SpectraST
```
and then the reader would have to deduce:
1. The current term names library creation software (amongst other things)
2. The current scope can accept library creation software
3. This term is therefore _the_ library creation software

This is not ideal if there's ambiguity. In this case it would be pretty contrived, but I'm not sure we can prove that universally.

Reply:
If the value is a CV term/accession, I think that should somehow be specified instead of simply allowing any string. But there also needs to be a way to denote that the term can't be used that way in mzML (e..g `<cvParam accession="MS:1003207" name="library creation software" value="MS:100147|SpectraST" />`). The only current way I can think to do that is to make sure the term doesn't map into mzML, by not making it a child of software. Which seems like a pretty yucky solution.

Reply:
I agree that we made a mistake by making this a free text value. We should go with a term without a value. That did not occur to me at the time. So I think we want to see this:
```
MS:1003207|library creation software=MS:1001477|SpectraST
```
in the format. so the revised entry should be this:
```
[Term]
id: MS:1003207
name: library creation software
def: "Library creation software" [PSI:MS]
is_a: MS:1000531 ! software
is_a: MS:1003171 ! spectral library attribute
```
i.e. no relationship: has_value_type.
and we should make sure that SpectraST and all other software terms that can create libraries are children of this term (not necessarily exclusively)

Does that seem agreeable?

Reply:
It doesn't express that the term can take a value, so one might not expect `MS:1003207|library creation software=MS:1001477|SpectraST` to be legal.

Another option, purely expedient and paying no heed to good form is to add a provision to the mzSpecLib specification that lets you use `<term1>=<term2>` statements where `term2` is an instance of `term1` to stipulate you are using `term2` in the role of `term1` explicitly. I am thinking of this like type casting in typed languages (`static_cast<char*>(stringvar)`, `(Base)Derived`). This gets us around that ambiguity problem, and leaves us free of recursive value type specifications.

Reply:
Isn't that what we are already doing? We already have all these in our examples:
```
MS:xxxxxxx|instrument model=MS:1000449|LTQ Orbitrap
MS:xxxxxxx|analyte type=MS:xxxxxxx|peptide
MS:1000525|spectrum representation=MS:1000127|centroid spectrum
MS:1000034|intensity unit=MS:1000905|percent of base peak times 100
MS:1000901|retention time normalization standard=MS:1002005|iRT retention normalization standard
MS:1000465|scan polarity=MS:1000130|positive scan
```

Reply:
You're right, I hadn't seen that. Removed the `has_value_type` clause.